es/es.go

```diff
@@ -202,6 +203,10 @@ func Run(ctx app.Context) {
 	}
 	etre.Debug("addr: %s", ctx.Options.Addr)
 
+	// Create a context with the timeout
+	ctxTimeout, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Shouldn't this be derived from the `ctx` parameter instead of `context.Background()`?